client/me/purchases/confirm-cancel-domain/index.jsx

```diff
@@ -135,22 +135,24 @@ class ConfirmCancelDomain extends React.Component {
 				return;
 			}
 
-			this.props.successNotice(
-				translate( '%(purchaseName)s was successfully cancelled and refunded.', {
-					args: { purchaseName },
-				} ),
-				{ displayOnNextPage: true }
-			);
+			if ( response.status === 'completed' ) {
+				this.props.successNotice(
+					translate( '%(purchaseName)s was successfully cancelled and refunded.', {
+						args: { purchaseName },
+					} ),
+					{ displayOnNextPage: true }
+				);
 
-			this.props.refreshSitePlans( purchase.siteId );
+				this.props.refreshSitePlans( purchase.siteId );
 
-			this.props.clearPurchases();
+				this.props.clearPurchases();
 
-			recordTracksEvent( 'calypso_domain_cancel_form_submit', {
-				product_slug: purchase.productSlug,
-			} );
+				recordTracksEvent( 'calypso_domain_cancel_form_submit', {
```

Comment:
When the status is marked as queued, for async plan deactivations, shouldn't we still record the event?

Reply:
Isn't that one for a future PR? For now you can't do async plan deactivation from calypso, even if the endpoints could.

Reply:
Yup, that will be handed in https://github.com/Automattic/wp-calypso/pull/55381.

Reply:
Got it. :)